File: C++/Linear_data_structure/stack.cpp
#include <iostream>
using namespace std;

#define STACK_SIZE 10 // Size of array

class Stack
{

private:
    int top;
    int stack_array[STACK_SIZE];

public:
    Stack()
    { // default constructor
        top = -1;
    }
    bool isEmpty()
    {
        if (top == -1)
            return true;
        else
            return false;
    }
    bool isFull()
    {
        if (top == STACK_SIZE - 1)
            return true;
        else
            return false;
    }
    void push(int data)
    {
        if (isFull())
        {
            cout << "\n\t Stack is overflow\n";
            return;
        }

        top = top + 1;
        stack_array[top] = data;
    }
    int peek()
    {
        if (isEmpty())
        {
            cout << "\n\t Stack is underflow\n";
            return -1;
        }
        return stack_array[top];
    }
    int pop()
    {
        if (isEmpty())
        {
            cout << "\n\t Stack is underflow\n";
            return -1;
        }
        return stack_array[top--];
    }

    void display()
    {
        if (isEmpty())
        {
            cout << "\n\t Stack is underflow\n";
            return;
        }

        cout << "\nStack : " << endl;
        for (int i = top; i >= 0; i--)
        {
            cout << "\t" << stack_array[i] << endl;
        }

        cout << endl;
    }
};

int main()
{
    Stack stack = Stack();

    stack.push(12);
    stack.push(20);
    stack.push(9);
    stack.push(17);
    stack.push(33);
    stack.display();
    cout << "\nPoped element is : " << stack.pop();
    stack.display();
    cout << "\nPoped element is : " << stack.pop();
    cout << "\nPoped element is : " << stack.pop();
    cout << "\nPoped element is : " << stack.pop();
    cout << "\nPoped element is : " << stack.pop();
    cout << "\nPoped element is : " << stack.pop();
    cout << "\nPoped element is : " << stack.pop();

    return 0;
}

File: java/src/stack_using_array/Stack.java
package stack_using_array;

public class Stack {
    private int stack[];
    final private int capacity;
    private int top;
    Stack(int capacity){
        this.capacity = capacity;
        this.stack = new int[capacity];
        this.top = -1;
    }
    private boolean isEmpty(){
        return this.top == -1;
    }
    private boolean isFull(){
        return this.top == this.capacity-1;
    }
    public void push(int data){
        if(this.isFull()){
            System.out.println("Stack is overflow");
            return;
        }

        this.top++;
        this.stack[this.top] = data;
    }
    public int pop(){
        if(this.isEmpty()){
            System.out.println("Stack is underflow");
            return -1;
        }
        return this.stack[this.top--];
    }
    public int peek(){
        if(this.isEmpty()){
            System.out.println("Stack is underflow");
            return -1;
        }
        return this.stack[this.top];
    }
    public void display(){
        if(this.isEmpty()){
            System.out.println("Stack is underflow");
            return;
        }
        for(int i = this.top; i>=0; i--){
            System.out.println(this.stack[i]);
        }
    }

    public int getSize(){
        return this.top+1;
    }
}
